classes/clean_moodle_url.php

```diff
@@ -357,6 +370,13 @@ public static function unclean($clean) {
                 self::log("Rewritten to: $path");
             }
 
+        } else if (preg_match("/^\/user\/(\w+)\/(\w+)$/", $path, $matches)) {
```

Comment:
Agree, we don't really know how many other usages are there for parame 'mode' . I hardcoded the above bit to expect exactly word 'discussions' in the end.